File: src/main/java/in/dragonbra/javasteam/steam/steamclient/callbackmgr/CallbackMsg.java
package in.dragonbra.javasteam.steam.steamclient.callbackmgr;

import in.dragonbra.javasteam.types.JobID;

/**
 * @author lngtr
 * @since 2018-02-22
 */
public class CallbackMsg implements ICallbackMsg {

    private JobID jobID = JobID.INVALID;

    @Override
    public JobID getJobID() {
        return this.jobID;
    }

    @Override
    public void setJobID(JobID jobID) {
        this.jobID = jobID;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/steamclient/callbackmgr/CallbackMsg.kt
package `in`.dragonbra.javasteam.steam.steamclient.callbackmgr

import `in`.dragonbra.javasteam.types.JobID

/**
 * Represents the base object all callbacks are based off.
 *
 * @constructor Initializes a new instance of the [CallbackMsg] class.
 *
 * @author lngtr
 * @since 2018-02-22
 */
abstract class CallbackMsg {

    /**
     * Gets or sets the job ID this callback refers to. If it is not a job callback, it will be [JobID.INVALID].
     */
    var jobID: JobID = JobID.INVALID
}
